torch/csrc/distributed/c10d/ProcessGroupNCCL.hpp

```diff
@@ -370,6 +370,9 @@ class TORCH_API ProcessGroupNCCL : public ProcessGroup {
   // may indicate that there is some sort of collective desynchronization.
   uint64_t getSequenceNumberForGroup() override;
 
+  // Tests if the UCC fallback path is available
+  bool isUCCAvailable() const;
+
```

Comment:
We may have to guard by `#ifdef USE_NCCL_WITH_UCC`, otherwise ld will error with 'undefined symbol: c10d::ProcessGroupNCCL::isUCCAvailable() const'

Reply:
I think this needs to be outside the `USE_NCCL_WITH_UCC` guard because in later PRs we need this API on the Python side to determine whether to run CPU tests on the "nccl" backend. So instead, I moved the definition of `isUCCAvailable` outside the guard:
```C++
bool ProcessGroupNCCL::isUCCAvailable() const {
#ifdef USE_NCCL_WITH_UCC
  return (uccPG_ != nullptr);
#else
  return false;
#endif
}
```